components/builder-depot/src/server.rs

```diff
@@ -1241,6 +1241,73 @@ fn list_package_versions(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+fn list_origin_packages_for_version(req: &mut Request) -> IronResult<Response> {
+
+    let (start, stop) = match extract_pagination(req) {
+        Ok(range) => range,
+        Err(response) => return Ok(response),
+    };
+
+    let ident = {
+        let params = req.extensions.get::<Router>().unwrap();
+        ident_from_params(params).to_string()
+    };
+
+    let packages: RouteResult<OriginPackagesForVersionResponse>;
+    let mut request = OriginPackagesForVersionRequest::new();
+    request.set_start(start as u64);
+    request.set_stop(stop as u64);
+
+    request.set_ident(OriginPackageIdent::from_str(ident.as_str()).expect(
```

Comment:
This is another one I pulled from elsewhere in this module, but in looking at it more closely now (and with slightly less bleary eyes), I'm questioning what it's doing, since the handler will always have been called with an origin, package and version, and `ident_from_params` will always return an `OriginPackageIdent`, which is what `set_ident` is expecting. Instead, I think I could probably get by with just this:

```
let ident = {
    let params = req.extensions.get::<Router>().unwrap();
    ident_from_params(params)
};
...
request.set_ident(ident);
```